core/src/history/events.rs

```diff
@@ -38,6 +41,7 @@ struct Value {
 
 #[derive(Debug, Default, Deserialize, Serialize)]
 pub struct EventRegistry {
+    version: Version<U1>,
```

Comment:
In my previous job we used epoch timestamps as version identifiers to avoid race conditions when two people bump the version with incompatible changes in parallel (leading e.g. to a v2 that is different from another v2 that has been merged to master before and thus not triggering an upgrade). 

Granted, we are not that many developers this might not be necessary here. But I also don't know if this typenum crate is really necessary here. Why not just use a u64?

Reply:
> But I also don't know if this typenum crate is really necessary here

It's not super necessary, but I think it works nicely here. So we have two options:
- Derialize/Serialize the `Version` and based on some constant. If we do that we need to have one implementation per struct we want to version (since they may need to use different version constants).
- Derialize/Serialize the `Version` and based on some type parameter. This way we can reuse `Version<V>` and just change the constant. Now, if rust had [const generics](https://github.com/rust-lang/rfcs/blob/master/text/2000-const-generics.md) we could write something like `Version<1>`, in the meantime, `typenum` provides a const-generic workaround.

I don't mind using a constant in this case, I just thought the `Version<V>` was pretty cool :smile: 

> In my previous job we used epoch timestamps as version identifiers to avoid race conditions when two people bump the version with incompatible changes in parallel

That's a very good point...

Reply:
Are you OK with merging this as is or should we try to use timestamp versioning?

Reply:
yes, please merge.